device-registry/src/main/scala/org/genivi/sota/device_registry/db/DeviceRepository.scala

```diff
@@ -72,10 +97,29 @@ object DeviceRepository {
       .filter(d => d.namespace === ns && d.deviceId === deviceId)
       .result
 
-  def search(ns: Namespace, re: String Refined Regex): DBIO[Seq[Device]] =
-    devices
-      .filter(d => d.namespace === ns && regex(d.deviceName, re))
-      .result
+  def search(ns: Namespace, re: String Refined Regex, offset: Option[Long], limit: Option[Long]): DBIO[Seq[Device]] =
```

Comment:
Agreed, however, it's not clear to me how to extract the filter code out. I had a few attempts and couldn't get anything that would compile.